src/libraries/System.Private.CoreLib/src/System/Globalization/TextInfo.cs

```diff
@@ -528,11 +538,13 @@ internal static void ToUpperAsciiInvariant(ReadOnlySpan<char> source, Span<char>
             }
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private static char ToLowerAsciiInvariant(char c)
         {
-            if ((uint)(c - 'A') <= (uint)('Z' - 'A'))
+            if (UnicodeUtility.IsInRangeInclusive(c, 'A', 'Z'))
             {
-                c = (char)(c | 0x20);
+                // on x86, extending BYTE -> DWORD is more efficient than WORD -> DWORD
```

Comment:
If this method is wriitten as
```c#
[MethodImpl(MethodImplOptions.AggressiveInlining)]
private static uint ToLowerAsciiInvariant1(uint c)
{
    if (UnicodeUtility.IsInRangeInclusive(c, 'A', 'Z'))
    {
        c |= 0x20;
    }
    return c;
}
```
so with `uint`s, then the `movzx` goes away entirely (even after inlining):
```diff
    L001c: cmp r10d, 0x19
    L0020: ja L002a
    L0022: or r8d, 0x20
-   L0026: movzx r8d, r8b
-   L002a: mov [r9], r8w
+   L002a: mov [r9], r8w
```
[Sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCUBXAOwwEt8YLAMIR8AB14AbGFADKMgG68wMXAG4aNYgGYmpBkIYBvGgzNNdzJExQMAYhAgAKWWOxcAPGAAW2KAD4GXDcuAEpTcxNqcxiGADNoBideHgZeBgBeBjo1NIYPIJCWABkYLgBzDG9cgGoa3nDo2LMo5ubg9wBtXgBdTIYAFQhiiAB3GQBBXDBeXgBJLgU/XncMJw6ubp7QjSbmgF8Is0O9hiOLJnJrYlsHCHIXEK9fAML3RpjWtoSoJJSMPJZHJ5AobEplSrVBh1BrnT5wtobLb9Jw+PyhIYjcZQKYzeaLZarB5I3o7BHHc4nGLnc6dACyMCqEAAJnNxJInAymaz2QB5MR8CBcXAsCblcqwXC4XgKGALSQpFLlUI9c5iKAy7AYGCXaxo36YsaTaazBZLDWrVEvBhgD6RclpOJJACqXGULJgzr4CowAE8WHNcAs2O5ynKuGBJBxpbLUWgGAByCYJ+MJgBaCdCduaXza5gA9PmGEKGAgABxIeMwBDarjMpUMABCAE0BgBRBhwQIAEQA6ry2N20rgGPhoDqYHE4speGUAVV3Ax+4POz3l92HTEwCj9aEnMBfdq99vCNkEKQ6GTTjEqW1iAB2G27G80070xneFlssQcrmfnk/vygrCqK4qSjG4YKm6FQqmqGpLNquoMBw/yDMMRo4ia+LmisPAPChqS2ucuaxLwTpOK67rMp63q8H6AZBlwIYVOGkbRjKMBxomyaphmWYOiReYnkC545A6t7NA+T6UjQJxaJYVx6AwlGQNRXpSHRvo0CR77ct+v4fl+fICrwQoimKEqqBB8qKjBqqnDoSHAI4kgMIGwahqxUYQU4BEAksUYwPGfkMJIGGNhA3DMsFqEcGIYgyBFUWhJkgROAFHA6nAoXhZFdYpR4WS+fFiV5cynY5diSX5bs+xAA)

Disadvantage is that on the call-site the cast `(char)` is necessary.

Reply:
That only helps if you mark `TextInfo.ToUpperInvariant(char)` with aggressive inlining, which I did not do here. The codegen is a bit too large to make that optimization worthwhile.

Reply:
The `uint`-suggestion is separate from `TextInfo.ToLowerInvariant(char)`'s inlining.

> Worst case it adds 1/4 nanosecond latency, so whatever. :)

Yeah :wink: